### PR TITLE
[Feature] Support LIP dataset for human parsing

### PR DESCRIPTION
**Pull request by MengzhangLI**
## Motivation

Old PR: https://github.com/open-mmlab/mmsegmentation/pull/1492

[LIP](https://arxiv.org/abs/1703.05446) is tasked for single human parsing. There are 19 semantic classes and 1 background class. The dataset is divided into 30K/10K/10K images for training, validation and testing. **In particular, training set has 30462 images and validation set has 10000 images.**

## Modification

1. Add mmseg/datasets/lip.py to get the images and labels.
2. Add LIPDataset in mmseg/datasets/init.py.
3. Add configs/_base_/datasets/lip.py for LIP dataset.
4. Add deeplabv3plus and pspnet config files of LIP dataset.

## Results

| Config                                   | mIoU img_scale = (520, 520)   | mIoU img_scale = (352, 352) | 
|------------------------------------------|---------| -------| 
| deeplabv3_r50-d8_321x321_80k_lip.py | 38.67 | **39.85** |
| deeplabv3_r50-d8_321x321_160k_lip.py | 39.70 | **39.89** |
| deeplabv3_r101-d8_321x321_80k_lip.py | 40.08 | **40.23** | 
| deeplabv3_r101-d8_321x321_160k_lip.py | **40.82** | 40.6 |
| deeplabv3plus_r50-d8_321x321_80k_lip.py | 40.14 | **41.61** |
| deeplabv3plus_r50-d8_321x321_160k_lip.py  | 40.95 | **41.9** |
| deeplabv3plus_r101-d8_321x321_80k_lip.py | **41.54** | 41.08 |
| deeplabv3plus_r101-d8_321x321_160k_lip.py| 41.77 | **42.31** |

| Config                                   | mIoU    |
|------------------------------------------|---------|
| ccnet_r50-d8_473x473_80k_lip.py | 40.73 |
| ccnet_r50-d8_473x473_160k_lip.py | 40.78 |
| ccnet_r101-d8_473x473_80k_lip.py | 41.76 |
| ccnet_r101-d8_473x473_160k_lip.py | 42.29 |
| deeplabv3plus_r50-d8_473x473_80k_lip.py | 42.02   |
| deeplabv3plus_r50-d8_473x473_160k_lip.py  | 42.86 |
| deeplabv3plus_r101-d8_473x473_80k_lip.py | 43.29 |
| deeplabv3plus_r101-d8_473x473_160k_lip.py| 43.69 |
| pspnet_r50-d8_473x473_80k_lip.py         | 40.85|
| pspnet_r50-d8_473x473_160k_lip.py        | 41.59 |
| pspnet_r101-d8_473x473_80k_lip.py        | 42.24   |
| pspnet_r101-d8_473x473_160k_lip.py       | 42.41 |

## Related reference 1:
https://arxiv.org/pdf/1811.11721.pdf
image

The crop size is also `(473, 473)`:

image

## Related reference 2:
https://arxiv.org/pdf/1804.01984.pdf
image

The crop size is also `(384, 384)` for JPPNet, `(321, 321)` for DeepLab and do inference on multi-scale inputs (with scales = 0.75, 0.5, 1.25) and also left-right flipped images. In particular, we compute as the final result the average probabilities from each scaleand flipped images, which is the same for predicting both parsing and pose.

image

## Next move (2022-09-06)

**with multi-scale inputs (with scales = 0.75, 0.5, 1.25) and also left-right flipped images.**

(1) ccnet_r50-d8_473x473_80k_lip.py
(2) ccnet_r50-d8_473x473_160k_lip.py
(3) ccnet_r101-d8_473x473_80k_lip.py
(4) ccnet_r101-d8_473x473_160k_lip.py

(5) deeplabv3plus_r50-d8_321x321_80k_lip.py
(6) deeplabv3plus_r50-d8_321x321_160k_lip.py
(7) deeplabv3plus_r101-d8_321x321_80k_lip.py
(8) deeplabv3plus_r101-d8_321x321_160k_lip.py
(9) deeplabv3plus_r101-d8_473x473_80k_lip.py
(10) deeplabv3plus_r101-d8_473x473_160k_lip.py

(11) deeplabv3_r50-d8_321x321_80k_lip.py
(12) deeplabv3_r50-d8_321x321_160k_lip.py
(13) deeplabv3_r101-d8_321x321_80k_lip.py
(14) deeplabv3_r101-d8_321x321_160k_lip.py

### CODE REVIEW

**Comment by MengzhangLI** [comment]
(1) deeplabv3plus_r50-d8_473x473_80k_lip.py
image

(2) deeplabv3plus_r50-d8_473x473_160k_lip.py
RUNNING

(3) pspnet_r50-d8_473x473_80k_lip.py
image

(4) pspnet_r50-d8_473x473_160k_lip.py
image

(5) pspnet_r101-d8_473x473_80k_lip.py
image
(6) pspnet_r101-d8_473x473_160k_lip.py 
RUNNING
